scipy/sparse/_sputils.py

```diff
@@ -241,21 +241,33 @@ def isintlike(x) -> bool:
     return True
 
 
-def isshape(x, nonneg=False, *, allow_1d=False) -> bool:
+def isshape(x, nonneg=False, *, allow_1d=False, allow_nd=False) -> bool:
     """Is x a valid tuple of dimensions?
 
     If nonneg, also checks that the dimensions are non-negative.
     If allow_1d, shapes of length 1 or 2 are allowed.
+    If allow_nd, shapes of all lengths >= 1 are allowed.
     """
     ndim = len(x)
-    if ndim != 2 and not (allow_1d and ndim == 1):
-        return False
-    for d in x:
-        if not isintlike(d):
+    if ndim < 3:
+        if ndim != 2 and not (allow_1d and ndim == 1):
             return False
-        if nonneg and d < 0:
+        for d in x:
+            if not isintlike(d):
+                return False
+            if nonneg and d < 0:
+                return False
+        return True
+    else:
+        if not allow_nd:
             return False
-    return True
+        for d in x:
```

Comment:
This block doesn't have to be duplicated from above. I think this function can be written as:
```python
    if ndim < 3:
        if ndim != 2 and not (allow_1d and ndim == 1):
            return False
    elif not allow_nd:
        return False
    else:
        for d in x:
            ...
```

Reply:
Addressed at 8ec8bd0.